includes/Checker/Preparations/Force_Single_Plugin_Preparation.php

```diff
@@ -74,39 +75,67 @@ public function prepare() {
 	}
 
 	/**
-	 * Filter active plugins.
+	 * Filters active plugins to only include required ones.
+	 *
+	 * This means:
+	 *
+	 * * The plugin being tested
+	 * * All dependencies of the plugin being tested
+	 * * Plugin Check itself
+	 * * All plugins depending on Plugin Check (they could be adding new checks)
 	 *
 	 * @since 1.0.0
 	 *
-	 * @param array $active_plugins List of active plugins.
-	 * @return array List of active plugins.
+	 * @param mixed $active_plugins List of active plugins.
+	 * @return mixed List of active plugins.
 	 */
 	public function filter_active_plugins( $active_plugins ) {
-		if ( is_array( $active_plugins ) && in_array( $this->plugin_basename, $active_plugins, true ) ) {
+		if ( ! is_array( $active_plugins ) ) {
+			return $active_plugins;
+		}
 
-			if ( defined( 'WP_PLUGIN_CHECK_MAIN_FILE' ) ) {
-				$plugin_check_file = WP_PLUGIN_CHECK_MAIN_FILE;
-			} else {
-				$plugins_dir       = defined( 'WP_PLUGIN_DIR' ) ? WP_PLUGIN_DIR : WP_CONTENT_DIR . '/plugins';
-				$plugin_check_file = $plugins_dir . '/plugin-check/plugin.php';
-			}
+		// The plugin being tested isn't actually active yet.
+		if ( ! in_array( $this->plugin_basename, $active_plugins, true ) ) {
+			return $active_plugins;
+		}
 
-			$plugin_base_file = plugin_basename( $plugin_check_file );
+		if ( defined( 'WP_PLUGIN_CHECK_MAIN_FILE' ) ) {
+			$plugin_check_file = WP_PLUGIN_CHECK_MAIN_FILE;
+		} else {
+			$plugins_dir       = defined( 'WP_PLUGIN_DIR' ) ? WP_PLUGIN_DIR : WP_CONTENT_DIR . '/plugins';
+			$plugin_check_file = $plugins_dir . '/plugin-check/plugin.php';
```

Comment:
Not strictly related to this PR since the code was already here before, but: Rather than doing this guesswork, wouldn't it be more reliable to look from here?
```suggestion
			$plugin_check_file = dirname( __DIR__, 3 ) . '/plugin.php';
```

Looks ugly, but should definitely be more reliable.

Reply:
Good point. Downside is that this needs to be kept in mind in case files are moved.